src/components/badges/BadgesOwnedForContract.tsx

```diff
@@ -1,24 +1,26 @@
-import { GoerliContractsStore } from 'stores/ContractsStore'
+import { BadgesContractsStore } from 'stores/ContractsStore'
 import { Suspense } from 'preact/compat'
 import { useSnapshot } from 'valtio'
 import BadgesOwnedForContractLoading from 'components/badges/BadgesOwnedForContractLoading'
 import MintedBadgeBlock from 'components/badges/MintedBadgeBlock'
 import WalletStore from 'stores/WalletStore'
+import useContractTokens from 'hooks/useContractTokens'
 
 function BadgesOwnedForContractSuspended({
   contractAddress,
 }: {
   contractAddress: string
 }) {
-  const { addressToTokenIds } = useSnapshot(GoerliContractsStore)
+  const ownedIds = useContractTokens(contractAddress, BadgesContractsStore)
   const { account } = useSnapshot(WalletStore)
+
   if (!account) {
     return <BadgesOwnedForContractLoading contractAddress={contractAddress} />
   }
-  if (!addressToTokenIds) {
+  if (ownedIds.length === 0) {
     return <BadgesOwnedForContractLoading contractAddress={contractAddress} />
```

Comment:
Seems like the return statement here is the same, as on `L18`, let's combine `if` info:
```TS
if (!account || ownedIds.length === 0) 
  return <BadgesOwnedForContractLoading contractAddress={contractAddress} />
```